test/unit/lib/serverless.test.js

```diff
@@ -174,4 +176,151 @@ describe('test/unit/lib/serverless.test.js', () => {
       expect(serverless.config).to.have.property('servicePath');
     });
   });
+
+  describe('Extend configuration', () => {
+    let serverless;
+    let parseEntriesStub;
+
+    before(async () => {
+      parseEntriesStub = sinon.stub().returns(new Map([]));
+      try {
+        await runServerless({
+          noService: true,
```

Comment:
I believe we need to run it in context of a _service_ to extend the service configuration

Reply:
As these are unit-tests they shouldn't depend on any services or package.  It's just the scope of the function what's being tested.

Reply:
In Framework there's a lot of cross-dependent machinery. So the way we ensure solid testing is by replicating full command runs. it's the only way to be sure that things will work as expected.

Reply:
This is the way we test things in this project. See: https://github.com/serverless/serverless/tree/main/test#unit-tests

Reply:
How am i supposed to use runServerless if it doesn't set/use the new internal variablesMeta? I'm just stuck on this.
Do i need to open a pull-request on @serverless/test as well?

Reply:
Sorry and please don't take that personally, but this is just madness to test a simple method like this.
I'm okay with running an integration test with an actual plugin but i refuse to unit-test a method this way.

Reply:
@medikoo 
Spawning the process means i cannot intercept/spy calls on related objects like variablesMeta. Means the tests would just loose quality in this case. I will make my thoughts on how to preserve quality and just update our local fork in the mean time.

Reply:
@mklenbw test should be constructed in the following way:
- Setup service with a plugin that updates the service configuration with new property values behind variables (as mentioned, you can rely on the already prepared fixture for that)
- Confirm by inspecting the stdout output of `sls print` command that it resolved the expected configuration.

There's no need for interception of anything. It's a simple end-to-end test that confirms that newly introduced functionality can be successfully used by the plugins

Reply:
So no testing of throw cases or proper cleanup of meta?
That would be a very shallow test.
But if it's that what you want i'll do it this way.